spark/interpreter/src/main/java/org/apache/zeppelin/spark/KotlinSparkInterpreter.java

```diff
@@ -0,0 +1,197 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import scala.Console;
+import org.apache.zeppelin.interpreter.BaseZeppelinContext;
+import org.apache.zeppelin.interpreter.Interpreter;
+import org.apache.zeppelin.interpreter.InterpreterContext;
+import org.apache.zeppelin.interpreter.InterpreterException;
+import org.apache.zeppelin.interpreter.InterpreterOutput;
+import org.apache.zeppelin.interpreter.InterpreterResult;
+import org.apache.zeppelin.interpreter.thrift.InterpreterCompletion;
+import org.apache.zeppelin.kotlin.KotlinInterpreter;
+import org.apache.zeppelin.spark.kotlin.KotlinZeppelinBindings;
+import org.apache.zeppelin.spark.kotlin.SparkKotlinReceiver;
+
+public class KotlinSparkInterpreter extends Interpreter {
+  private static Logger logger = LoggerFactory.getLogger(KotlinSparkInterpreter.class);
+  private static final SparkVersion KOTLIN_SPARK_SUPPORTED_VERSION = SparkVersion.SPARK_2_4_0;
+
+  private KotlinInterpreter interpreter;
+  private SparkInterpreter sparkInterpreter;
+  private BaseZeppelinContext z;
+  private JavaSparkContext jsc;
+  private SparkVersion sparkVersion;
+
+  public KotlinSparkInterpreter(Properties properties) {
+    super(properties);
+    logger.debug("Creating KotlinSparkInterpreter");
+    interpreter = new KotlinInterpreter(properties);
+  }
+
+  @Override
+  public void open() throws InterpreterException {
+    sparkInterpreter =
+        getInterpreterInTheSameSessionByClassName(SparkInterpreter.class);
+    jsc = sparkInterpreter.getJavaSparkContext();
+
+    sparkVersion = SparkVersion.fromVersionString(jsc.version());
+    assertVersion();
+
+    z = sparkInterpreter.getZeppelinContext();
+
+    SparkKotlinReceiver ctx = new SparkKotlinReceiver(
+        sparkInterpreter.getSparkSession(),
+        jsc,
+        sparkInterpreter.getSQLContext(),
+        z);
+
+    List<String> classpath = sparkClasspath();
+
+    String outputDir = null;
+    SparkConf conf = jsc.getConf();
+    if (conf != null) {
+      outputDir =  conf.getOption("spark.repl.class.outputDir").getOrElse(null);
+    }
+
+    interpreter.properties()
+        .receiver(ctx)
+        .classPath(classpath)
+        .outputDir(outputDir)
+        .codeOnLoad(KotlinZeppelinBindings.Z_SELECT_KOTLIN_SYNTAX)
+        .codeOnLoad(KotlinZeppelinBindings.SPARK_UDF_IMPORTS)
+        .codeOnLoad(KotlinZeppelinBindings.CAST_SPARK_SESSION);
+    interpreter.open();
+  }
+
+  @Override
+  public void close() throws InterpreterException {
+    interpreter.close();
+  }
+
+  @Override
+  public InterpreterResult interpret(String st, InterpreterContext context)
+      throws InterpreterException {
+    assertVersion();
```

Comment:
for instance, in interpret, it will be much nicer to return the message in InterpreterResult than throwing exception

Reply:
Fixed, now error InterpreterResult is returned with message about actual and supported version